packages/common/src/utils.js

```diff
@@ -457,3 +458,20 @@ export function defineAlias(src, target, prop, opts = {}) {
     }
   })
 }
+
+export function serializeWithMethods(obj, name) {
```

Comment:
A final change request 🙈 Why not calling it `serializeFunction`? (And also ensuring first arg `fn` is an instance of function)

Reply:
PS: We may infer the function name using `fn.name` :)  (Maybe an extra replace to ensure the name is stripped off)

Reply:
done! :)